aws/resource_aws_eks_node_group.go

```diff
@@ -470,15 +471,15 @@ func resourceAwsEksNodeGroupUpdate(d *schema.ResourceData, meta interface{}) err
 		output, err := conn.UpdateNodegroupVersion(input)
 
 		if err != nil {
-			return fmt.Errorf("error updating EKS Node Group (%s) version: %w", d.Id(), err)
+			return diag.Errorf("error updating EKS Node Group (%s) version", d.Id(), err)
 		}
 
 		updateID := aws.StringValue(output.Update.Id)
 
-		_, err = waiter.NodegroupUpdateSuccessful(conn, clusterName, nodeGroupName, updateID, d.Timeout(schema.TimeoutUpdate))
+		_, err = waiter.NodegroupUpdateSuccessful(ctx, conn, clusterName, nodeGroupName, updateID, d.Timeout(schema.TimeoutUpdate))
 
 		if err != nil {
-			return fmt.Errorf("error waiting for EKS Node Group (%s) version update (%s): %w", d.Id(), updateID, err)
+			return diag.Errorf("error waiting for EKS Node Group (%s) version update (%s)", d.Id(), updateID, err)
```

Comment:
Still need to pass `error`, just can't use `%w`.

```suggestion
			return diag.Errorf("error waiting for EKS Node Group (%s) version update (%s): %s", d.Id(), updateID, err)
```